app/src/main/java/edu/uw/myuw_android/CommonWebViewFragment.kt

```diff
@@ -70,16 +100,21 @@ class CommonWebViewFragment: Fragment() {
         super.onViewCreated(view, savedInstanceState)
 
         Log.d("CommonWebViewFragment", "Title: ${args.title}")
+        Log.d("CommonWebViewFragment", "unique_id: ${args.uniqueId}")
+
+        baseUrl = resources.getString(R.string.myuw_base_url)
 
-        if (!webViewMap.containsKey(args.title)) {
-            webViewMap[args.title] = WebView(view.context)
-            webViewMap[args.title]!!.layoutParams = ViewGroup.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.MATCH_PARENT)
+        if (!webViewMap.containsKey(args.uniqueId)) {
+            webViewMap[args.uniqueId] = WebView(view.context)
+            webViewMap[args.uniqueId]!!.layoutParams = ViewGroup.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.MATCH_PARENT)
```

Comment:
That value has to non null because it is assigned in the same scope with a constructor. If it is null then the app has to crash because it means that a constructor failed and there is no work around that. It is not there to check if the value is null but to assure the compiler that the value is not null. Changing it to '?' is bad style in my opinion because then it will let a unrecoverable state pass along to other parts of the code.

Reply:
What about other !!s? Maybe use as few "!!" as possible?